Tests/L0/SonarQubePostTest/PRCA/PostCommentsTests.ps1

```diff
@@ -81,7 +106,69 @@ namespace Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi
         public CommentType CommentType { get; set; }
         public string Content { get; set; }
         public bool IsDeleted { get; set; }
+        
+        public int DiscussionId { get; set; }
     }
+    
+    public class DiscussionHttpClient
+    {
+        // The mock implementation keeps a list of threads that were posted and uses it to get requests
+        List<DiscussionThread> postedThreads = new List<DiscussionThread>();
+        int lastDiscussionId = 1;
+        
+        // For test - to be able to easily get to the threads that were posted
+        public List<DiscussionThread> GetPostedThreads()
+        {
+            return postedThreads;
+        }
+        
+        // This mock implementation simply records the threads that are posted to the PR
+        // It also rewrites the discussion IDs as the server would do and links the comment's discussion id to the parent thread discussion id
+        public Task<DiscussionThreadCollection> CreateThreadsAsync(VssJsonCollectionWrapper<DiscussionThreadCollection> newThreads, object userState = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            TaskCompletionSource<DiscussionThreadCollection> tsc = new TaskCompletionSource<DiscussionThreadCollection>();
+            DiscussionThreadCollection threads = newThreads.Value as DiscussionThreadCollection;
+            
+            foreach (var thread in threads)
+            {
+                thread.DiscussionId = lastDiscussionId++;
```

Comment:
The module creates threads and assigns negative numbers to their ID. When the threads get to the server these are persisted and the IDs are changed. Comments are also updated to have the proper discussion ID. The mock emulates this server side functionality.